### ISSUE TITLE
Using \AtEveryPend{\vspace{x}} adds empty line (same for \AtEveryPstart, \AtEveryStanza)

### ISSUE THREAD

**Issue by tweh** [maieul/ledmac#610]
As said on the mailing list using `\AtEveryPend{\vspace{x}}` doesn’t work as expected. It seems like it always inserts an empty line ind normal numbered and parallel text. But only in normal numbered text we can use a negative value to shift back to lets say a resulting distance of `0.5\baselineskip`. In parallel typesetting the distance is always at least one baseline skip.

It is the same for `\AtEveryPstart` and `\AtEveryStanza`.

Example:
```latex
\documentclass{article}

\usepackage{reledmac,reledpar}

\AtEveryPend{\vspace{0mm}}
\AtEveryPend{\vspace{-6.5mm}}

\begin{document}

Line of normal text

\beginnumbering
\pstart First Paragraph\pend
\pstart Second Paragraph\pend
\endnumbering

Line of normal text

\bigskip
\hrule
\bigskip

Line of normal text

\begin{pairs}
   \begin{Leftside}
      \beginnumbering
      \pstart First Paragraph\pend
      \pstart Second Paragraph\pend
      \endnumbering
   \end{Leftside}
   \begin{Rightside}
      \beginnumbering
      \pstart Erster Absatz\pend
      \pstart Zweiter Absatz\pend
      \endnumbering
   \end{Rightside}
\end{pairs}
\Columns

Line of normal text

\end{document}
```

If we try to use `\AtEveryStanza` to add some custom numbering or just a symbol at the stanzas start we se that this dos to it’s own line and is not – as I’d expect ’ prepended directly to the first line of the stanza:

Example
```latex
\documentclass{article}

\usepackage{reledmac,reledpar}

\setstanzaindents{1,0}
\setcounter{stanzaindentsrepetition}{1}

\usepackage{amssymb}

\AtEveryStanza{$\bigstar$\quad}% a star in the beginning of the first stanza line

\begin{document}

Line of normal text

\beginnumbering
\stanza First Line & Second line\&
\stanza Third Line & Fourth line\&
\endnumbering

Line of normal text

\bigskip
\hrule
\bigskip

Line of normal text

\begin{pairs}
   \begin{Leftside}
      \beginnumbering
      \stanza First Line & Second line \&
      \stanza Third Line & Fourth line \&
      \endnumbering
   \end{Leftside}
   \begin{Rightside}
      \beginnumbering
      \stanza Erste Zeile & Zweite Zeile \&
      \stanza Dritte Zeile & Vierte Zeile \&
      \endnumbering
   \end{Rightside}
\end{pairs}
\Columns

Line of normal text

\end{document}
```

**Comment by maieul** [comment]
notes perso:
- premier probl�me: \AtEveryPstart / \AtEveryPend insert un \noindent� dans la majorit� des cas c'est justifi�, mais il faudrait une variante pour ne pas l'ins�rer
- scond probl�me: en colonne on utilise des \parbox� je sais plus la riason, mais il y en a une

**Comment by tweh** [comment]
Another note: It works if we use

```latex
\usepackage{etoolbox}
\apptocmd{\stanza}{$\bigstar$\quad}{}{}% a star in the beginning of the first stanza line
```

instead

**Comment by maieul** [comment]
that is logical� but I don't think it is good idea (etoolbox is already called by reledmac)

**Comment by tweh** [comment]
Wha isn’t it a good idea?

**Comment by maieul** [comment]
because 
1) patching command in general is not a good idea 
2) that should be solve directly in reledmac
3) I am note sure WHERE exactly the space is inserted relativly to all what a \stanza do�
-----
Ma�eul
http://blog.maieul.net
http://www.schtroumpfs.maieul.net/

Le 12 janv. 2017 � 20:47, Tobias Weh <notifications@github.com> a �crit :

**Comment by tweh** [comment]
Well, I’d prefer a reledmac/par internal solution too, but for now its a workaround I’m ok with :-)

**Comment by maieul** [comment]
Note: that is the expected behavior for verse. It was concepted to insert vertical material between lines of verses / verses. But I will add new tools to insert automatically something inside pstart/pend.

**Comment by maieul** [comment]
Please try branch issue610. It provides:
- starred command of AtEveryXXX which does not insert \noindent (your first problem)
- \AtStartEveryStanza, \BeforeEveryStopStanza, \AtEndEveryPend, \AtStartEveryPstart. which inserts content not between lines, but inside lines (your seconde problem).

**Comment by maieul** [comment]
hi, some news?

**Comment by tweh** [comment]
Hi,

> \AtStartEveryStanza, \BeforeEveryStopStanza, \AtEndEveryPend, \AtStartEveryPstart. which inserts content not between lines, but inside lines (your seconde problem).

This works fine.

> starred command of AtEveryXXX which does not insert \noindent (your first problem)

This doesn’t work or at least I didn’t understand it … place take a look at the following example and the tables.

**old (non-starred) macros**

|                       | BLS | Text | parallel Text | Verses | parallel Verses |
|----------------------|-----|------|---------------|--------|-----------------|
| `\AtEveryPstart`     | 0   |❌  >EL|   ❌  >EL     |❌  >EL|        ❌  >EL   |
| `\AtEveryPstart`     | 0.25|❌  >EL|   ❌  >EL     |❌  >EL|        ❌  >EL   |
| `\AtEveryPend`       | 0   |❌  >EL|   ❌  >EL     |❌  >EL|        ❌  >EL   |
| `\AtEveryPend`       | 0.25|❌  >EL|   ❌  >EL     |❌  >EL|        ❌  >EL   |
| `\AtEveryStanza`     | 0   |  —   |   —           |❌  >EL|        ❌  >EL   |
| `\AtEveryStanza`     | 0.25|  —   |   —           |❌  >EL|        ❌  >EL   |
| `\AtEveryStopStanza` | 0   |  —   |   —           |❌  >EL|        ❌  >EL   |
| `\AtEveryStopStanza` | 0.25|  —   |   —           |❌  >EL|        ❌  >EL   |

**new (starred) macors**

|                       | BLS | Text | parallel Text | Verses | parallel Verses |
|-----------------------|-----|------|---------------|--------|-----------------|
| `\AtEveryPstart*`     | 0   |  ✅  |   ❌  >EL     |   ✅   |        ❌  >EL   |
| `\AtEveryPstart*`     | 0.25|  ✅  |   ❌  >EL     |   ✅   |        ❌  >EL   |
| `\AtEveryPend*`       | 0   |  ✅  |   ❌  >EL     |   ✅   |        ❌  >EL   |
| `\AtEveryPend*`       | 0.25|  ✅  |   ❌  >EL     |   ✅   |        ❌  >EL   |
| `\AtEveryStanza*`     | 0   |  —   |   —           |   ✅   |        ❌  >EL   |
| `\AtEveryStanza*`     | 0.25|  —   |   —           |   ✅   |        ❌  >EL   |
| `\AtEveryStopStanza*` | 0   |  —   |   —           |   ✅   |        ❌  >EL   |
| `\AtEveryStopStanza*` | 0.25|  —   |   —           |   ✅   |        ❌  >EL   |

- *BLS* = amount of `\baselineskip` in the `\vspace` argument
- ✅ = works as expected
- ❌ = doesn’t work as expected, i.e. inserts more vertical space (at least `1\baselineskip`)
- *>EL* = at least one empty line is inserted


**Code**

```tex
\documentclass{article}

\usepackage{reledmac,reledpar}

\setstanzaindents{1,0}
\setcounter{stanzaindentsrepetition}{1}

%\AtEveryPstart{\vspace{0\baselineskip}}
%\AtEveryPend{\vspace{0\baselineskip}}
%\AtEveryStanza{\vspace{0\baselineskip}}
%\AtEveryStopStanza{\vspace{0\baselineskip}}

%\AtEveryPstart{\vspace{0.25\baselineskip}}
%\AtEveryPend{\vspace{0.25\baselineskip}}
%\AtEveryStanza{\vspace{0.25\baselineskip}}
%\AtEveryStopStanza{\vspace{0.25\baselineskip}}

%\AtEveryPstart*{\vspace{0\baselineskip}}
%\AtEveryPend*{\vspace{0\baselineskip}}
%\AtEveryStanza*{\vspace{0\baselineskip}}
%\AtEveryStopStanza*{\vspace{0\baselineskip}}

%\AtEveryPstart*{\vspace{0.25\baselineskip}}
%\AtEveryPend*{\vspace{0.25\baselineskip}}
%\AtEveryStanza*{\vspace{0.25\baselineskip}}
%\AtEveryStopStanza*{\vspace{0.25\baselineskip}}

\begin{document}

\section*{Text}

Line of normal text

\beginnumbering
\pstart First Paragraph\pend
\pstart Second Paragraph\pend
\endnumbering

Line of normal text

\section*{Parallel Text}

Line of normal text

\begin{pairs}
   \begin{Leftside}
      \beginnumbering
      \pstart First Paragraph\pend
      \pstart Second Paragraph\pend
      \endnumbering
   \end{Leftside}
   \begin{Rightside}
      \beginnumbering
      \pstart Erster Absatz\pend
      \pstart Zweiter Absatz\pend
      \endnumbering
   \end{Rightside}
\end{pairs}
\Columns

Line of normal text

\section*{Verses}

Line of normal text

\beginnumbering
\stanza First Line & Second line\&
\stanza Third Line & Fourth line\&
\endnumbering

Line of normal text

\section*{Parallel Verses}

Line of normal text

\begin{pairs}
   \begin{Leftside}
      \beginnumbering
      \stanza First Line & Second line \&
      \stanza Third Line & Fourth line \&
      \endnumbering
   \end{Leftside}
   \begin{Rightside}
      \beginnumbering
      \stanza Erste Zeile & Zweite Zeile \&
      \stanza Dritte Zeile & Vierte Zeile \&
      \endnumbering
   \end{Rightside}
\end{pairs}
\Columns

\end{document}
```

**Comment by maieul** [comment]
I ommited to test with parallel column, I did it only for parallel pages. I will try to solve this issue.

**Comment by maieul** [comment]
hum, that is a complex  issue…

**Comment by maieul** [comment]
please, try the new version of the branch. Not that for starred version, in parallel column, there is no difference between left and right side. So if you do `\AtEveryPstart{\vspace{0.25\baselineskip}}, you will obtain 0.5\baselineskip.

**Comment by tweh** [comment]
Hm … the spacing seems to work but I get this error message:

```
! You can't use a prefix with `\endgroup'.
```

> So if you do `\AtEveryPstart{\vspace{0.25\baselineskip}}`, you will obtain `0.5\baselineskip`.

Can this be prevented. I think this is a bit strange behaviour …

**Comment by maieul** [comment]
ok, I will look as soon as possible.

**Comment by tweh** [comment]
thanks. But I’m not in a hurry. At the moment I don’t have a project using reledmac/-par. So you can take a look, when you have the time :-)

**Comment by maieul** [comment]
> 
> Can this be prevented. I think this is a bit strange behaviour …


That could, but this will insert new exception, and maybe boarding effect. That is a "normal" behavior: you have two \pstart (one for left, one for right) at the same emplacement in the page, so that is  logical you get it twice.

If I want to prevent this, I need to add only the optional argument for one of the pstart… and that will be add more desillusion for people.

You can use this "hack" to avoid this problem

````
\AtEveryPstart*{%
\ifl@dprintingcolumns
  \vspace{0.125\baselineskip}
\else
  \vspace{0.25\baselineskip}
\fi
}
\makeatother
`````

cf 
872d737

**Comment by tweh** [comment]
Hi, the new version seems to work fine :-)

> If I want to prevent this, I need to add only the optional argument for one of the pstart… and that will be add more desillusion for people.

I don't know the internals but for me typesetting in sync an adding a vertical space it seems un-logic that the space is inserted twice. But I can live with it, it should be stated in the manual, I think …

**Comment by maieul** [comment]
That is not a problem of internal. 
You told "AtEveryPstart"-> there is two pstart, one by side. So it is normal the content AtEveryPstart is typeset two times.

**Comment by tweh** [comment]
I see, but I won’t consider this as *tow* start but rather *one* start in *two* languages ;-)

**Comment by maieul** [comment]
You type two\pstart, you have two \pstart… that is logical.

**Comment by maieul** [comment]
what about if the AtEveryPstart contain not vspace but real content?

**Comment by tweh** [comment]
Yes … I see what you mean and understand the logic :-) However it feels strange, that the vertical space get inserted twice. But as I said it would be fine if it is mentioned in the manual.

Maybe a better solution would be make `\parskip` work with `reledmac` and add an equivalent length `\stanzaskip`.

**Comment by maieul** [comment]
I don't think it is a good idea, because it will add more non-generic command.

**Comment by maieul** [comment]
please tell me what you think about the change of the handbook on 160ba83

**Comment by tweh** [comment]
Yes I thinks the manual text is fine.

**Comment by maieul** [comment]
Cool. I prepare the release. I wait for an English corrector…

**Comment by tweh** [comment]
👍

**Comment by AbuJordan** [comment]
Maieul,
I've tried your code for this and checked the manual and I still can't get it to work, can you publish a full example
here's my code
\documentclass{book}
\usepackage[a4paper]{geometry}
\usepackage{hyperref}
\usepackage{polyglossia}
\newcommand{\LuaLaTeX}{Lua\LaTeX}

\usepackage[series={},nocritical,noend,nofamiliar,noledgroup]{reledmac}
\usepackage{reledpar} 

\linenumincrement*{1}
\firstlinenum*{1}
\setlength{\Lcolwidth}{0.60\textwidth}
\setlength{\Rcolwidth}{0.35\textwidth} 
\usepackage{libertine}
\setmainlanguage{english}
\setotherlanguage{hebrew}    
\newfontfamily\hebrewfont[Script=Hebrew]{Times New Roman}

\begin{document}

\begin{pairs}
\AtEveryPstart*{%
\ifl@dprintingcolumns
  \vspace{0.125\baselineskip}
\else
  \vspace{0.25\baselineskip}
\fi
}
\begin{Leftside} 
\begin{RTL}
\begin{hebrew}
\beginnumbering
\pstart
\eledchapter{text here}\ledleftnote{s}
\pend
\pstart
\large

אחרי מלחמת העולם השנייה היה הדבר פרופ ציחק בער, אחד מגדולי חוקרי תולדות ישׂראל,הפקדי בידי את מפעל ההוצאה הביקורתית של ספר יוסיפון, על סמך כתבי-היד וקטעי הגניזה אשר שרדו לפליטה, וזכות נעימה לי להביע בזה את תודתי על 
\pend    
\endnumbering
\end{hebrew}
\end{RTL}
\end{Leftside}


\begin{Rightside} 
\begin{english}
\beginnumbering
\pstart
\eledchapter{text here 2}   
\pend 
\pstart
\scriptsize
Adam Seth Enosh Kenan Mehalalel Jared Enoch Methusaleh Lamech Noah.
Noah was the father of Shem, Ham and Japheth.
The sons of Japheth were: Gomer, Magog, Madai, Javan, Tubal, Meshech, and Tiras.
\pend
\endnumbering
\end{english}
\end{Rightside}
\end{pairs}
\Columns
\end{document}

**Comment by maieul** [comment]
@AbuJordan please open a new issue, with a detailled explanation of the problem

**Comment by AbuJordan** [comment]
I don't think it's a new issue, I'm trying to use the work around that you wrote, but I'm wondering about syntax, where it should be put and if there needs to be any packages loaded
````
\AtEveryPstart*{%
\ifl@dprintingcolumns
  \vspace{0.125\baselineskip}
\else
  \vspace{0.25\baselineskip}
\fi
}
\makeatother
`````
